storm-server/src/main/java/org/apache/storm/scheduler/blacklist/strategies/DefaultBlacklistStrategy.java

```diff
@@ -85,9 +85,10 @@ public void prepare(Map<String, Object> conf) {
             }
         }
         Set<String> toRelease = releaseBlacklistWhenNeeded(cluster, new ArrayList<>(blacklist.keySet()));
+        cluster.setQuarantinedSupervisors(toRelease);
```

Comment:
I don't like this to be a side effect of this `getBlacklist` function. But I am still looking for a better way to better resolve this.

Reply:
I apologize for taking so long to update on this. I have been busy on other things. 

I couldn't find a better way to solve this problem without breaking backwards compatibility of `IBlacklistStrategy` interface.  But to make the code more clear, here are my suggestions:

1. don't use `quarantined`. The meaning of `quarantined` can be confusing, especially under the context here, we are not saying these nodes can't be used. They are just lowest priority. I think `graylist` is a better word in this context. `blacklist` means don't use it, `graylist` means don't use it unless have to.
2. Add comments about this side effect (updating graylist) in this `getBlacklist` function

We should refactor the blacklistScheduler code in the future if possible.